### PR TITLE
Change _reward_fn to take action taken

### LINKED ISSUE #155: Allow reward function to consider action taken
## 🚀  Feature: Allow reward function to consider action taken

It would be great if `_reward_fn` could take in the action taken, as well as the previous and current observation. This would enable users to define custom reward functions that take actions, which is common in RL literature. The context in which `_reward_fn` is called (`env.step(action)`) has this action in scope.

I think it would be relatively straightforward to implement by changing a few lines in `nle/env/base.py` and `nle/env/tasks.py`.

### PR DESCRIPTION
**Pull request by RobertKirk**
In standard RL often the reward function takes as input the action
taken. This change allows that to happen by adjusting _reward_fn.

See issue https://github.com/facebookresearch/nle/issues/155 for more details

Closes #155

### CODE REVIEW

File: nle/env/tasks.py [hunk 1]
```diff
@@ -194,7 +196,7 @@ def __init__(self, *args, **kwargs):
 
         super().__init__(*args, options=options, **kwargs)
 
-    def _reward_fn(self, last_observation, observation, end_status):
+    def _reward_fn(self, last_observation, action, observation, end_status):
         """Difference between previous gold and new gold."""
         del end_status  # Unused
```
Comment (heiner) [inline nle/env/tasks.py:201]: Thanks! To keep this in line with how this code was written so far, would you mind adding `del action` lines to make linters happy about `action` not being used in these functions at this point?

**Review by heiner — COMMENTED** [review]
Looks good, short comment below.

File: nle/env/tasks.py (hunk 1 above)
Comment (RobertKirk) [reply nle/env/tasks.py:202]: Sure thing, I've made that change

**Review by heiner — APPROVED** [review]
Awesome, thanks!

**Pull request merged**